sdk/core/azure_core_amqp/README.md

```diff
@@ -1,4 +1,4 @@
-<!-- cspell: words EVENTHUB eventhubs -->
+<!-- cspell: words  -->
```

Comment:
Nit: just get rid of this now and delete the initial empty lines so the H1 is first.

Reply:
Yes, the eh_consumer.md file is essentially a README for the module. 

Somehow I convinced GH copilot to create this and I honestly decided I really liked it (because the module level documentation is incredibly sparse). 

So rather than cut and paste the entire goo as a rustdoc comment, I used include_str to include it into the module documentation.

I'm rather happy with how it renders :).

Reply:
I like the idea, but to standardize around something, what if we make the module a directory with its own README, to keep everything nice and tight together i.e.,

```
src/
  consumer/
    README.md
    mod.rs
  lib.rs
```

That also has the benefit of rendering in the GitHub web UI.